### ISSUE TITLE
Allow to manage two instances with same data dir

### ISSUE THREAD

**Issue by ov7a** [elastic/puppet-elasticsearch#807]
## Feature Description

Currently if one specifies two instanses with same data dir:

    
    elasticsearch::instance {es1
         datadir => '/ssd/elasticsearch',
         config => $es_config,
         init_defaults => $init_defaults
    }
    elasticsearch::instance {es2:
        datadir => '/ssd/elasticsearch',
        config => $es_config2,
        init_defaults => $init_defaults,
    }
    
Puppet will give an error like this

> ... Duplicate declaration: File[/ssd/elasticsearch] is already declared in file /etc/puppetlabs/code/environments/production/modules/elasticsearch/manifests/instance.pp:352; cannot redeclare at /etc/puppetlabs/code/environments/production/modules/elasticsearch/manifests/instance.pp:352 at /etc/puppetlabs/code/environments/production/modules/elasticsearch/manifests/instance.pp:352:5 ...

But it seems it is ok to have one data dir for several instances because it has a number for instance (data_dir/cluster_name/nodes/0/).

It would be nice to have an option to manage it externally or to detect such duplication.

A workaround is to comment `File[$instance_datadir]` declaration.

**Comment by tylerjl** [comment]
Hi @ov7a, does using the `datadir` parameter on the parent `elasticsearch` class fix that problem for you? For example,

```puppet
class { 'elasticsearch':
  datadir => '/ssd/elasticsearch',
}

(instance definitions for es1 and es2)
```

Will manage the instance data directories as you describe.

**Comment by ov7a** [comment]
Unfortunately, no. It changes `path.data` parameter from `/ssd/elasticsearch` to `/ssd/elasticsearch/es1` / `/ssd/elasticsearch/es2`. In other words, the data stored on different paths.

**Comment by tylerjl** [comment]
Ah, I see what you're saying. Unfortunately what you're describing shouldn't be done; the internal `path.data` directory structure that you're talking about (`nodes/0`/etc.) should be local to the instance only, and multiple instances of elasticsearch should _not_ share the exact same path.data directory (see [this answer](http://stackoverflow.com/a/23501459) and [this howto](https://discuss.elastic.co/t/can-i-run-multiple-elasticsearch-nodes-on-the-same-machine/67)). Although the top-level `datadir` parameter I described doesn't get you exactly the path layout you describe, you'll still get the same effect (storing all your ES data in one top-level directory).

Let me know if this answers your question.

**Comment by ov7a** [comment]
Yep, it would be nice. 
Please, clarify your question about workaround. The "comment datadir declaration" workaround works for me.

**Comment by ov7a** [comment]
For the record: https://www.elastic.co/guide/en/elasticsearch/reference/5.3/modules-node.html#max-local-storage-nodes

**Comment by tylerjl** [comment]
Implemented this in #826, support will be in the next release.
